### PR TITLE
fix: install all packages from versions.json also when using pnpm

### LINKED ISSUE #12834: Platform frontend dependencies are not added in pnpm mode when no vaadin dependency
### Description of the bug

Platform [version.json](https://github.com/vaadin/platform/blob/master/versions.json) file defines the frontend dependencies which are added by Flow since https://github.com/vaadin/flow/commit/ef7c5a36aa46bd91ff3a8c14fc032419b8b40573. However, it only works for `npm` mode, not for `pnpm`. we should make the behavior consistent.

Besides, adding frontend dependencies defined by the platform has the benefit for Hilla, that a Hilla user will get all the platform defined dependencies automatically without having to have the `vaadin` dependency (which pulls the server-side Java components that are not needed for Hilla)

### Expected behavior

Platform dependencies are added when running in both npm and pnpm mode

### Minimal reproducible example

https://github.com/Artur-/hilla-test

Note that the platform dependencies are added in npm mode, but not pnpm

### Versions

- Vaadin / Flow version:
- Java version:
- OS version:
- Browser version (if applicable):
- Application Server (if applicable):
- IDE (if applicable):

### PR DESCRIPTION
**Pull request by haijian-vaadin**
## Description

Please list all relevant dependencies in this section and provide summary of the change, motivation and context.

Fixes #12834
## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

**Review by Artur- — DISMISSED** [review]

**Comment by Artur-** [comment]
The summary and description should be changed to reflect what this really does: Installs all packages mentioned in the versions json in addition to all `@NpmPackages`

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/TaskUpdatePackagesNpmTest.java [hunk 1]
```diff
@@ -198,23 +200,19 @@ public void npmIsInUse_noPlatformVersionJsonPresent_noFailure()
...
+    @Test
+    public void pnpmIsInUse_platformVersionsJsonAdded_dependenciesAdded()
+            throws IOException {
+        verifyPlatformDependenciesAreAdded(true);
+    }
 
-        verifyVersions(newVersion, newVersion, newVersion);
-        verifyVersionLockingWithNpmOverrides(true, true, true);
+    @Test
+    public void npmIsInUse_platformVersionsJsonAdded_dependenciesAdded()
```
Comment (Artur-) [inline flow-server/src/test/java/com/vaadin/flow/server/frontend/TaskUpdatePackagesNpmTest.java:213]: How does this test differ from `npmIsInUse_platformVersionsJsonAdded_versionsPinned` ?

Comment (haijian-vaadin) [reply flow-server/src/test/java/com/vaadin/flow/server/frontend/TaskUpdatePackagesNpmTest.java:229]: Maybe you reviewed a few seconds too early, now the two tests are different. For example, the new tests have empty application dependencies, just to verify the platform dependencies are added.

Comment (haijian-vaadin) [reply flow-server/src/test/java/com/vaadin/flow/server/frontend/TaskUpdatePackagesNpmTest.java:229]: `npmIsInUse_platformVersionsJsonAdded_versionsPinned` also run the task twice with different conditions. The new tests just verify the dependencies are added, nothing else.

**Review by Artur- — APPROVED** [review]

**Pull request merged**